packages/taquito-remote-signer/src/auth.ts

```diff
@@ -0,0 +1,54 @@
+import { checkDecodeTezosID, encodeTezosID } from '@taquito/michel-codec';
+import * as ed25519 from '@stablelib/ed25519';
+import * as blake2b from '@stablelib/blake2b';
+import elliptic from 'elliptic';
+import { TezosIDType } from '@taquito/michel-codec';
+
+enum PublicKeyHashID {
+    ED25519 = 0,
+    SECP256K1 = 1,
+    P256 = 2,
+}
+
+function computeDigest(msg: Uint8Array, pkh: [TezosIDType, number[]]): Uint8Array {
+    const hashType = pkh[0] == 'ED25519PublicKeyHash' ? PublicKeyHashID.ED25519 :
+        pkh[0] == 'SECP256K1PublicKeyHash' ? PublicKeyHashID.SECP256K1 :
+        PublicKeyHashID.P256;
+
+    const buf = new Uint8Array(msg.length + pkh[1].length + 3);
+    buf.set([4, 1, hashType]);
+    buf.set(pkh[1], 3);
+    buf.set(msg, 3+pkh[1].length);
+    return blake2b.hash(buf, 32);
+}
+
+export function authenticateRequest(msg: Uint8Array, pk: string, pkh: string): string {
```

Comment:
The parameter `pk` can be confused with public key, should we use `sk` instead?

Reply:
What about `secret_key`? `msg` and `pkh` are unambiguous, and used comprehensively, so there is "precedent" for those...